File: extensions/panache/hibernate-reactive-panache/deployment/src/test/java/io/quarkus/hibernate/reactive/panache/test/MyOtherTestResource.java
package io.quarkus.hibernate.reactive.panache.test;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.PathParam;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.WebApplicationException;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;

import io.smallrye.mutiny.Uni;

@Path("other-entity")
public class MyOtherTestResource {

    @GET
    @Path("{id}")
    @Produces(MediaType.APPLICATION_JSON)
    public Uni<MyOtherEntity> get(@PathParam("id") long id) {
        return MyOtherEntity.<MyOtherEntity> findById(id)
                .onItem().ifNull().failWith(() -> new WebApplicationException(Response.Status.NOT_FOUND));
    }
}

File: extensions/panache/hibernate-reactive-panache-kotlin/deployment/src/test/kotlin/io/quarkus/hibernate/reactive/panache/kotlin/test/MyOtherTestResource.kt
package io.quarkus.hibernate.reactive.panache.kotlin.test

import io.smallrye.mutiny.Uni
import java.util.function.Supplier
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("other-entity")
class MyOtherTestResource {
    @GET
    @Path("{id}")
    @Produces(MediaType.APPLICATION_JSON)
    fun get(@PathParam("id") id: Long): Uni<MyOtherEntity> {
        return MyOtherEntity
            .findById(id)
            .onItem().ifNull().failWith { WebApplicationException(Response.Status.NOT_FOUND) }
    }
}